src/lightning_lite/plugins/collectives/collective.py

```diff
@@ -0,0 +1,137 @@
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional
+
+import torch
+from typing_extensions import Self
+
+from lightning_lite.utilities.types import CollectibleGroup
+
+
+class Collective(ABC):
```

Comment:
A simple docstring would be nice, something like:

A thin wrapper around torch process group for backend agnostic collective operations.

Reply:
Done